langlib/lang.regexp/src/main/java/org/ballerinalang/langlib/regexp/Find.java

```diff
@@ -75,6 +96,12 @@ public static BArray findAll(BRegexpValue regExp, BString str, int startIndex) {
     }
 
     public static BArray findAllGroups(BRegexpValue regExp, BString str, int startIndex) {
+        int strLength = str.length();
```

Comment:
All of the methods have similar code sample for this check. Can't we wrap this into a separate method?

Reply:
Added in https://github.com/ballerina-platform/ballerina-lang/pull/39238/commits/30d5dd4058fc549010a92ea234b4ede20d3998b4